### PR TITLE
Onc 39 integrate react into extension

### PR DESCRIPTION
**Pull request by ApplebaumIan**
## Important changes
- All view communication logic from `media/main.js` has been moved to `views/src/App.tsx` 
- To run the app in development mode run `yarn watch` then in VSCode hit the `F5` key. 
    - refreshing VSCode will reload the web view's changes.
- To build the app for sharing `yarn compile`

## Changes
- the application's views are now built with React
- [MUI Library is installed](https://mui.com/material-ui/getting-started/) for easy prebuilt UI components.

### CODE REVIEW

**Review by IshanAggarwal125 — APPROVED** [review]

**Review by IshanAggarwal125 — APPROVED** [review]
The code looks good to me.

**Review by IshanAggarwal125 — APPROVED** [review]

**Review by jbbernardin — APPROVED** [review]
The only question/concern I have is in getNonce, it appears that we are saving the authentication password in the code itself without any encryption, which seems like poor practice (I could also be misunderstanding the purpose of this function). To be honest though, even if this is the case, it's probably fine for our usage,

**Comment by ApplebaumIan** [comment]
@jbbernardin So actually the `getNonce()` function actually doesn't save authentication. It's actually for cross side scripting purposes as we are importing our JS code into the VSCode web view. Basically the `getNonce()` is run once and creates a unique identifier for the web view, we set the script to also share this UID so that it can call it. If you were to remove this you actually get a slew of JS errors and it will appear as if the extension doesn't work at all (speaking from experience 😅)

File: src/extension.ts [hunk 1]
```diff
@@ -367,16 +369,21 @@ class ChatGPTViewProvider implements vscode.WebviewViewProvider {
...
+export function deactivate() {}
+
+function getNonce() {
+	let text = "";
+	const possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+	for (let i = 0; i < 32; i++) {
+		text += possible.charAt(Math.floor(Math.random() * possible.length));
+	}
+	return text;
+}
```
Comment (ApplebaumIan) [inline src/extension.ts:389]: @jbbernardin basically this function looks at all the possible characters and randomly chooses 32 of them to create the unique identifier.

**Pull request merged**